==13771== Memcheck, a memory error detector
==13771== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==13771== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==13771== Command: /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory
==13771== 
==13771== Conditional jump or move depends on uninitialised value(s)
==13771==    at 0x68E7CD: wasmparser::validator::Validator::end (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x6842CC: wasmparser::validator::Validator::payload (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x683541: wasmparser::validator::Validator::validate_all (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x29CA0C: wasmer_compiler::compiler::Compiler::validate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2680A8: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::validate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x57F954: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25E8A1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25CCEA: memory::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25EB36: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x730902: {{closure}} (rt.rs:52)
==13771==    by 0x730902: do_call<closure-0,i32> (panicking.rs:348)
==13771==    by 0x730902: try<i32,closure-0> (panicking.rs:325)
==13771==    by 0x730902: catch_unwind<closure-0,i32> (panic.rs:394)
==13771==    by 0x730902: std::rt::lang_start_internal (rt.rs:51)
==13771==    by 0x25E3A1: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==  Uninitialised value was created by a stack allocation
==13771==    at 0x29C920: wasmer_compiler::compiler::Compiler::validate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== Warning: set address range perms: large range [0x59e43000, 0x1d9e43000) (noaccess)
==13771== 
==13771== HEAP SUMMARY:
==13771==     in use at exit: 32,434 bytes in 114 blocks
==13771==   total heap usage: 1,016 allocs, 902 frees, 183,558 bytes allocated
==13771== 
==13771== 6 bytes in 1 blocks are still reachable in loss record 1 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x66087F: wasmer_compiler::translator::environ::ModuleEnvironment::declare_memory_export (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x65D3AA: wasmer_compiler::translator::sections::parse_export_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x669B5D: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x65FB08: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x286FE4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x26825A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x57F9AF: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25E8A1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25CCEA: memory::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25EB36: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x730902: {{closure}} (rt.rs:52)
==13771==    by 0x730902: do_call<closure-0,i32> (panicking.rs:348)
==13771==    by 0x730902: try<i32,closure-0> (panicking.rs:325)
==13771==    by 0x730902: catch_unwind<closure-0,i32> (panic.rs:394)
==13771==    by 0x730902: std::rt::lang_start_internal (rt.rs:51)
==13771== 
==13771== 8 bytes in 1 blocks are still reachable in loss record 2 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x2E715A: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2E36F0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E64EB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x28F6D1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x296275: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x29AA28: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2B6471: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2A2EA1: <wasmer_compiler_cranelift::compiler::CraneliftCompiler as wasmer_compiler::compiler::Compiler>::compile_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2874C8: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x26825A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 12 bytes in 1 blocks are still reachable in loss record 3 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x664B40: alloc::raw_vec::RawVec<T,A>::reserve_exact (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x65C136: wasmer_compiler::translator::sections::parse_function_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x6695D4: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x65FB08: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x286FE4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x26825A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x57F9AF: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25E8A1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25CCEA: memory::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25EB36: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x730902: {{closure}} (rt.rs:52)
==13771==    by 0x730902: do_call<closure-0,i32> (panicking.rs:348)
==13771==    by 0x730902: try<i32,closure-0> (panicking.rs:325)
==13771==    by 0x730902: catch_unwind<closure-0,i32> (panic.rs:394)
==13771==    by 0x730902: std::rt::lang_start_internal (rt.rs:51)
==13771== 
==13771== 16 bytes in 1 blocks are still reachable in loss record 4 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x664850: alloc::raw_vec::RawVec<T,A>::reserve_exact (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x65C657: wasmer_compiler::translator::sections::parse_memory_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x669AD5: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x65FB08: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x286FE4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x26825A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x57F9AF: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25E8A1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25CCEA: memory::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25EB36: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x730902: {{closure}} (rt.rs:52)
==13771==    by 0x730902: do_call<closure-0,i32> (panicking.rs:348)
==13771==    by 0x730902: try<i32,closure-0> (panicking.rs:325)
==13771==    by 0x730902: catch_unwind<closure-0,i32> (panic.rs:394)
==13771==    by 0x730902: std::rt::lang_start_internal (rt.rs:51)
==13771== 
==13771== 16 bytes in 2 blocks are still reachable in loss record 5 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x665931: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x65B644: wasmer_compiler::translator::sections::parse_type_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x6698EF: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x65FB08: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x286FE4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x26825A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x57F9AF: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25E8A1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25CCEA: memory::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25EB36: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x730902: {{closure}} (rt.rs:52)
==13771==    by 0x730902: do_call<closure-0,i32> (panicking.rs:348)
==13771==    by 0x730902: try<i32,closure-0> (panicking.rs:325)
==13771==    by 0x730902: catch_unwind<closure-0,i32> (panic.rs:394)
==13771==    by 0x730902: std::rt::lang_start_internal (rt.rs:51)
==13771== 
==13771== 16 bytes in 2 blocks are still reachable in loss record 6 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x665931: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x65B5EF: wasmer_compiler::translator::sections::parse_type_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x6698EF: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x65FB08: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x286FE4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x26825A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x57F9AF: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25E8A1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25CCEA: memory::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25EB36: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x730902: {{closure}} (rt.rs:52)
==13771==    by 0x730902: do_call<closure-0,i32> (panicking.rs:348)
==13771==    by 0x730902: try<i32,closure-0> (panicking.rs:325)
==13771==    by 0x730902: catch_unwind<closure-0,i32> (panic.rs:394)
==13771==    by 0x730902: std::rt::lang_start_internal (rt.rs:51)
==13771== 
==13771== 16 bytes in 1 blocks are indirectly lost in loss record 7 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x262F52: wasmer_engine::tunables::Tunables::create_memories (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x283419: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x57FD0D: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x585502: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25CE2F: memory::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25EB36: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x730902: {{closure}} (rt.rs:52)
==13771==    by 0x730902: do_call<closure-0,i32> (panicking.rs:348)
==13771==    by 0x730902: try<i32,closure-0> (panicking.rs:325)
==13771==    by 0x730902: catch_unwind<closure-0,i32> (panic.rs:394)
==13771==    by 0x730902: std::rt::lang_start_internal (rt.rs:51)
==13771==    by 0x25E3A1: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 20 bytes in 3 blocks are still reachable in loss record 8 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x6606BF: wasmer_compiler::translator::environ::ModuleEnvironment::declare_func_export (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x65D38B: wasmer_compiler::translator::sections::parse_export_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x669B5D: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x65FB08: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x286FE4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x26825A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x57F9AF: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25E8A1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25CCEA: memory::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25EB36: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x730902: {{closure}} (rt.rs:52)
==13771==    by 0x730902: do_call<closure-0,i32> (panicking.rs:348)
==13771==    by 0x730902: try<i32,closure-0> (panicking.rs:325)
==13771==    by 0x730902: catch_unwind<closure-0,i32> (panic.rs:394)
==13771==    by 0x730902: std::rt::lang_start_internal (rt.rs:51)
==13771== 
==13771== 20 bytes in 3 blocks are still reachable in loss record 9 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x660B3F: wasmer_compiler::translator::environ::ModuleEnvironment::declare_function_name (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x65EB32: wasmer_compiler::translator::sections::parse_name_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x669D50: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x65FB08: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x286FE4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x26825A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x57F9AF: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25E8A1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25CCEA: memory::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25EB36: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x730902: {{closure}} (rt.rs:52)
==13771==    by 0x730902: do_call<closure-0,i32> (panicking.rs:348)
==13771==    by 0x730902: try<i32,closure-0> (panicking.rs:325)
==13771==    by 0x730902: catch_unwind<closure-0,i32> (panic.rs:394)
==13771==    by 0x730902: std::rt::lang_start_internal (rt.rs:51)
==13771== 
==13771== 24 bytes in 1 blocks are indirectly lost in loss record 10 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x266446: <alloc::boxed::Box<[T]> as core::clone::Clone>::clone (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2838AA: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x57FD0D: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x585502: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25CE2F: memory::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25EB36: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x730902: {{closure}} (rt.rs:52)
==13771==    by 0x730902: do_call<closure-0,i32> (panicking.rs:348)
==13771==    by 0x730902: try<i32,closure-0> (panicking.rs:325)
==13771==    by 0x730902: catch_unwind<closure-0,i32> (panic.rs:394)
==13771==    by 0x730902: std::rt::lang_start_internal (rt.rs:51)
==13771==    by 0x25E3A1: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 24 bytes in 1 blocks are indirectly lost in loss record 11 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x266356: <alloc::boxed::Box<[T]> as core::clone::Clone>::clone (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2838D7: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x57FD0D: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x585502: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25CE2F: memory::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25EB36: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x730902: {{closure}} (rt.rs:52)
==13771==    by 0x730902: do_call<closure-0,i32> (panicking.rs:348)
==13771==    by 0x730902: try<i32,closure-0> (panicking.rs:325)
==13771==    by 0x730902: catch_unwind<closure-0,i32> (panic.rs:394)
==13771==    by 0x730902: std::rt::lang_start_internal (rt.rs:51)
==13771==    by 0x25E3A1: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 32 bytes in 1 blocks are still reachable in loss record 12 of 48
==13771==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x4E3F7F4: _dlerror_run (dlerror.c:140)
==13771==    by 0x4E3F165: dlsym (dlsym.c:70)
==13771==    by 0x73EED3: fetch (weak.rs:61)
==13771==    by 0x73EED3: get<unsafe extern "C" fn(*const libc::unix::linux_like::linux::gnu::b64::x86_64::pthread_attr_t) -> usize> (weak.rs:46)
==13771==    by 0x73EED3: min_stack_size (thread.rs:449)
==13771==    by 0x73EED3: std::sys::unix::thread::Thread::new (thread.rs:49)
==13771==    by 0x2E01F6: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E3576: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E412F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E7126: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2E36F0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E64EB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x28F6D1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 40 bytes in 1 blocks are indirectly lost in loss record 13 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x6A86EE: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x6B2ED6: wasmer_vm::memory::LinearMemory::new_internal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x6B2C18: wasmer_vm::memory::LinearMemory::from_definition (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x582A01: <wasmer::tunables::Tunables as wasmer_engine::tunables::Tunables>::create_vm_memory (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x26303B: wasmer_engine::tunables::Tunables::create_memories (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x283419: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x57FD0D: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x585502: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25CE2F: memory::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25EB36: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x730902: {{closure}} (rt.rs:52)
==13771==    by 0x730902: do_call<closure-0,i32> (panicking.rs:348)
==13771==    by 0x730902: try<i32,closure-0> (panicking.rs:325)
==13771==    by 0x730902: catch_unwind<closure-0,i32> (panic.rs:394)
==13771==    by 0x730902: std::rt::lang_start_internal (rt.rs:51)
==13771== 
==13771== 64 bytes in 4 blocks are still reachable in loss record 14 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x2E975C: crossbeam_deque::deque::Worker<T>::new_lifo (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2EA5D7: core::iter::traits::iterator::Iterator::unzip (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E3A06: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E7126: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2E36F0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E64EB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x28F6D1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x296275: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x29AA28: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2B6471: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 64 bytes in 4 blocks are still reachable in loss record 15 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x73EE2C: alloc (alloc.rs:80)
==13771==    by 0x73EE2C: alloc (alloc.rs:174)
==13771==    by 0x73EE2C: exchange_malloc (alloc.rs:277)
==13771==    by 0x73EE2C: std::sys::unix::thread::Thread::new (thread.rs:44)
==13771==    by 0x2E01F6: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E3576: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E412F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E7126: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2E36F0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E64EB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x28F6D1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x296275: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x29AA28: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 72 bytes in 1 blocks are still reachable in loss record 16 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x64600D: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x649C7B: wasmer_engine::trap::frame_info::register (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x289585: <wasmer_engine_jit::artifact::JITArtifact as wasmer_engine::artifact::Artifact>::register_frame_info (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x283865: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x57FD0D: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x585502: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25CE2F: memory::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25EB36: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x730902: {{closure}} (rt.rs:52)
==13771==    by 0x730902: do_call<closure-0,i32> (panicking.rs:348)
==13771==    by 0x730902: try<i32,closure-0> (panicking.rs:325)
==13771==    by 0x730902: catch_unwind<closure-0,i32> (panic.rs:394)
==13771==    by 0x730902: std::rt::lang_start_internal (rt.rs:51)
==13771==    by 0x25E3A1: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 88 bytes in 1 blocks are possibly lost in loss record 17 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x658912: hashbrown::raw::inner::RawTable<T>::fallible_with_capacity (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x65832E: _ZN9hashbrown3raw5inner17RawTable$LT$T$GT$14reserve_rehash17he29be5fc862d84d8E.llvm.8588116468508767673 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x65D4C1: wasmer_compiler::translator::sections::parse_export_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x669B5D: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x65FB08: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x286FE4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x26825A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x57F9AF: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25E8A1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25CCEA: memory::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25EB36: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 128 bytes in 4 blocks are still reachable in loss record 18 of 48
==13771==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x56C48E1: __cxa_thread_atexit_impl (cxa_thread_atexit_impl.c:106)
==13771==    by 0x70B45E: register_dtor (fast_thread_local.rs:33)
==13771==    by 0x70B45E: try_register_dtor<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>> (local.rs:438)
==13771==    by 0x70B45E: std::thread::local::fast::Key<T>::try_initialize (local.rs:424)
==13771==    by 0x72DC59: get<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,fn() -> core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>> (local.rs:410)
==13771==    by 0x72DC59: __getit (local.rs:175)
==13771==    by 0x72DC59: try_with<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,closure-0,()> (local.rs:262)
==13771==    by 0x72DC59: with<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,closure-0,()> (local.rs:239)
==13771==    by 0x72DC59: std::sys_common::thread_info::set (thread_info.rs:40)
==13771==    by 0x2E037C: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x73F249: call_once<(),FnOnce<()>> (boxed.rs:1081)
==13771==    by 0x73F249: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==13771==    by 0x73F249: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==13771==    by 0x52516DA: start_thread (pthread_create.c:463)
==13771==    by 0x57A271E: clone (clone.S:95)
==13771== 
==13771== 128 bytes in 4 blocks are still reachable in loss record 19 of 48
==13771==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x56C48E1: __cxa_thread_atexit_impl (cxa_thread_atexit_impl.c:106)
==13771==    by 0x2F1E2D: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E9902: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E1206: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E4E49: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E2F5A: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E59F0: _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hd13678e4d803656bE.llvm.9549009445676040077 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E03BE: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x73F249: call_once<(),FnOnce<()>> (boxed.rs:1081)
==13771==    by 0x73F249: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==13771==    by 0x73F249: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==13771==    by 0x52516DA: start_thread (pthread_create.c:463)
==13771==    by 0x57A271E: clone (clone.S:95)
==13771== 
==13771== 128 bytes in 1 blocks are indirectly lost in loss record 20 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x582AF4: <wasmer::tunables::Tunables as wasmer_engine::tunables::Tunables>::create_vm_memory (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x26303B: wasmer_engine::tunables::Tunables::create_memories (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x283419: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x57FD0D: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x585502: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25CE2F: memory::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25EB36: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x730902: {{closure}} (rt.rs:52)
==13771==    by 0x730902: do_call<closure-0,i32> (panicking.rs:348)
==13771==    by 0x730902: try<i32,closure-0> (panicking.rs:325)
==13771==    by 0x730902: catch_unwind<closure-0,i32> (panic.rs:394)
==13771==    by 0x730902: std::rt::lang_start_internal (rt.rs:51)
==13771==    by 0x25E3A1: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 144 bytes in 1 blocks are still reachable in loss record 21 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x664A47: alloc::raw_vec::RawVec<T,A>::reserve_exact (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x65B3F7: wasmer_compiler::translator::sections::parse_type_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x6698EF: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x65FB08: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x286FE4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x26825A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x57F9AF: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25E8A1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25CCEA: memory::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25EB36: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x730902: {{closure}} (rt.rs:52)
==13771==    by 0x730902: do_call<closure-0,i32> (panicking.rs:348)
==13771==    by 0x730902: try<i32,closure-0> (panicking.rs:325)
==13771==    by 0x730902: catch_unwind<closure-0,i32> (panic.rs:394)
==13771==    by 0x730902: std::rt::lang_start_internal (rt.rs:51)
==13771== 
==13771== 152 bytes in 1 blocks are still reachable in loss record 22 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x667EEF: hashbrown::raw::RawTable<T>::reserve_rehash (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x668292: hashbrown::raw::RawTable<T>::insert (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x6574A4: hashbrown::map::HashMap<K,V,S>::insert (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x660BC4: wasmer_compiler::translator::environ::ModuleEnvironment::declare_function_name (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x65EB32: wasmer_compiler::translator::sections::parse_name_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x669D50: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x65FB08: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x286FE4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x26825A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x57F9AF: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25E8A1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 160 bytes in 4 blocks are still reachable in loss record 23 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x2E73EC: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E4A4B: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2EA838: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E5980: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E3AAA: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E7126: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2E36F0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E64EB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x28F6D1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x296275: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 160 bytes in 4 blocks are still reachable in loss record 24 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x2E73EC: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E4A76: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2EA838: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E5980: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E3AAA: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E7126: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2E36F0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E64EB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x28F6D1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x296275: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 160 bytes in 4 blocks are still reachable in loss record 25 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x2E73EC: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2EA9FB: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E3B20: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E7126: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2E36F0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E64EB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x28F6D1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x296275: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x29AA28: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2B6471: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 160 bytes in 4 blocks are still reachable in loss record 26 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x70C7FE: alloc (alloc.rs:80)
==13771==    by 0x70C7FE: alloc (alloc.rs:174)
==13771==    by 0x70C7FE: exchange_malloc (alloc.rs:277)
==13771==    by 0x70C7FE: new<()> (mutex.rs:223)
==13771==    by 0x70C7FE: std::thread::Thread::new (mod.rs:1149)
==13771==    by 0x2E00E1: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E3576: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E412F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E7126: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2E36F0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E64EB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x28F6D1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x296275: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x29AA28: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 160 bytes in 4 blocks are still reachable in loss record 27 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x2E0103: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E3576: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E412F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E7126: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2E36F0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E64EB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x28F6D1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x296275: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x29AA28: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2B6471: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 192 bytes in 4 blocks are still reachable in loss record 28 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x7298F6: alloc (alloc.rs:80)
==13771==    by 0x7298F6: alloc (alloc.rs:174)
==13771==    by 0x7298F6: exchange_malloc (alloc.rs:277)
==13771==    by 0x7298F6: std::sync::condvar::Condvar::new (condvar.rs:130)
==13771==    by 0x2E4A56: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2EA838: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E5980: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E3AAA: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E7126: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2E36F0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E64EB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x28F6D1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x296275: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 192 bytes in 4 blocks are still reachable in loss record 29 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x7298F6: alloc (alloc.rs:80)
==13771==    by 0x7298F6: alloc (alloc.rs:174)
==13771==    by 0x7298F6: exchange_malloc (alloc.rs:277)
==13771==    by 0x7298F6: std::sync::condvar::Condvar::new (condvar.rs:130)
==13771==    by 0x2E4A81: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2EA838: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E5980: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E3AAA: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E7126: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2E36F0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E64EB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x28F6D1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x296275: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 192 bytes in 4 blocks are still reachable in loss record 30 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x7298F6: alloc (alloc.rs:80)
==13771==    by 0x7298F6: alloc (alloc.rs:174)
==13771==    by 0x7298F6: exchange_malloc (alloc.rs:277)
==13771==    by 0x7298F6: std::sync::condvar::Condvar::new (condvar.rs:130)
==13771==    by 0x2EAA01: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E3B20: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E7126: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2E36F0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E64EB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x28F6D1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x296275: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x29AA28: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2B6471: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 192 bytes in 4 blocks are still reachable in loss record 31 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x70C862: alloc (alloc.rs:80)
==13771==    by 0x70C862: alloc (alloc.rs:174)
==13771==    by 0x70C862: exchange_malloc (alloc.rs:277)
==13771==    by 0x70C862: new (condvar.rs:130)
==13771==    by 0x70C862: std::thread::Thread::new (mod.rs:1150)
==13771==    by 0x2E00E1: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E3576: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E412F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E7126: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2E36F0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E64EB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x28F6D1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x296275: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x29AA28: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 280 bytes in 1 blocks are still reachable in loss record 32 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x664947: alloc::raw_vec::RawVec<T,A>::reserve_exact (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x65D213: wasmer_compiler::translator::sections::parse_export_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x669B5D: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x65FB08: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x286FE4: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x26825A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x57F9AF: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25E8A1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25CCEA: memory::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25EB36: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x730902: {{closure}} (rt.rs:52)
==13771==    by 0x730902: do_call<closure-0,i32> (panicking.rs:348)
==13771==    by 0x730902: try<i32,closure-0> (panicking.rs:325)
==13771==    by 0x730902: catch_unwind<closure-0,i32> (panic.rs:394)
==13771==    by 0x730902: std::rt::lang_start_internal (rt.rs:51)
==13771== 
==13771== 320 bytes in 4 blocks are still reachable in loss record 33 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x70C8B1: alloc (alloc.rs:80)
==13771==    by 0x70C8B1: alloc (alloc.rs:174)
==13771==    by 0x70C8B1: exchange_malloc (alloc.rs:277)
==13771==    by 0x70C8B1: new<std::thread::Inner> (sync.rs:324)
==13771==    by 0x70C8B1: std::thread::Thread::new (mod.rs:1145)
==13771==    by 0x2E00E1: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E3576: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E412F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E7126: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2E36F0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E64EB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x28F6D1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x296275: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x29AA28: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 384 bytes in 1 blocks are still reachable in loss record 34 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x2E8C0C: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E593C: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E3AAA: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E7126: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2E36F0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E64EB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x28F6D1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x296275: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x29AA28: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2B6471: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 416 bytes in 4 blocks are still reachable in loss record 35 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x2E0198: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E3576: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E412F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E7126: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2E36F0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E64EB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x28F6D1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x296275: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x29AA28: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2B6471: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 512 bytes in 1 blocks are still reachable in loss record 36 of 48
==13771==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x72F369: aligned_malloc (alloc.rs:95)
==13771==    by 0x72F369: alloc (alloc.rs:22)
==13771==    by 0x72F369: __rdl_alloc (alloc.rs:304)
==13771==    by 0x2E8F26: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E3ADA: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E7126: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2E36F0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E64EB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x28F6D1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x296275: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x29AA28: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 640 bytes in 1 blocks are still reachable in loss record 37 of 48
==13771==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x72F369: aligned_malloc (alloc.rs:95)
==13771==    by 0x72F369: alloc (alloc.rs:22)
==13771==    by 0x72F369: __rdl_alloc (alloc.rs:304)
==13771==    by 0x2E3D16: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E7126: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2E36F0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E64EB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x28F6D1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x296275: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x29AA28: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2B6471: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 640 bytes in 1 blocks are still reachable in loss record 38 of 48
==13771==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x72F369: aligned_malloc (alloc.rs:95)
==13771==    by 0x72F369: alloc (alloc.rs:22)
==13771==    by 0x72F369: __rdl_alloc (alloc.rs:304)
==13771==    by 0x2F2232: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2F1ECD: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E9902: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E1206: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E4E49: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E2F5A: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E59F0: _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hd13678e4d803656bE.llvm.9549009445676040077 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E03BE: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 664 bytes in 1 blocks are still reachable in loss record 39 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x287353: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x26825A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x57F9AF: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25E8A1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25CCEA: memory::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25EB36: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x730902: {{closure}} (rt.rs:52)
==13771==    by 0x730902: do_call<closure-0,i32> (panicking.rs:348)
==13771==    by 0x730902: try<i32,closure-0> (panicking.rs:325)
==13771==    by 0x730902: catch_unwind<closure-0,i32> (panic.rs:394)
==13771==    by 0x730902: std::rt::lang_start_internal (rt.rs:51)
==13771==    by 0x25E3A1: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 664 (432 direct, 232 indirect) bytes in 1 blocks are definitely lost in loss record 40 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x6ADA9D: wasmer_vm::instance::InstanceHandle::allocate_instance (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x283231: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x57FD0D: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x585502: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25CE2F: memory::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25EB36: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x730902: {{closure}} (rt.rs:52)
==13771==    by 0x730902: do_call<closure-0,i32> (panicking.rs:348)
==13771==    by 0x730902: try<i32,closure-0> (panicking.rs:325)
==13771==    by 0x730902: catch_unwind<closure-0,i32> (panic.rs:394)
==13771==    by 0x730902: std::rt::lang_start_internal (rt.rs:51)
==13771==    by 0x25E3A1: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 808 bytes in 1 blocks are still reachable in loss record 41 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x65156F: alloc::collections::btree::map::BTreeMap<K,V>::insert (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x64A081: wasmer_engine::trap::frame_info::register (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x289585: <wasmer_engine_jit::artifact::JITArtifact as wasmer_engine::artifact::Artifact>::register_frame_info (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x283865: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x57FD0D: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x585502: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25CE2F: memory::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x25EB36: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x730902: {{closure}} (rt.rs:52)
==13771==    by 0x730902: do_call<closure-0,i32> (panicking.rs:348)
==13771==    by 0x730902: try<i32,closure-0> (panicking.rs:325)
==13771==    by 0x730902: catch_unwind<closure-0,i32> (panic.rs:394)
==13771==    by 0x730902: std::rt::lang_start_internal (rt.rs:51)
==13771==    by 0x25E3A1: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 1,152 bytes in 4 blocks are possibly lost in loss record 42 of 48
==13771==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x4013666: allocate_dtv (dl-tls.c:286)
==13771==    by 0x4013666: _dl_allocate_tls (dl-tls.c:530)
==13771==    by 0x5252227: allocate_stack (allocatestack.c:627)
==13771==    by 0x5252227: pthread_create@@GLIBC_2.2.5 (pthread_create.c:644)
==13771==    by 0x73EF7C: std::sys::unix::thread::Thread::new (thread.rs:66)
==13771==    by 0x2E01F6: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E3576: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E412F: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E7126: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2E36F0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E64EB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x28F6D1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 1,520 bytes in 1 blocks are still reachable in loss record 43 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x2E3B72: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E7126: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2E36F0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E64EB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x28F6D1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x296275: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x29AA28: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2B6471: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2A2EA1: <wasmer_compiler_cranelift::compiler::CraneliftCompiler as wasmer_compiler::compiler::Compiler>::compile_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2874C8: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 1,536 bytes in 4 blocks are still reachable in loss record 44 of 48
==13771==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x72F369: aligned_malloc (alloc.rs:95)
==13771==    by 0x72F369: alloc (alloc.rs:22)
==13771==    by 0x72F369: __rdl_alloc (alloc.rs:304)
==13771==    by 0x2E9800: crossbeam_deque::deque::Worker<T>::new_lifo (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2EA5D7: core::iter::traits::iterator::Iterator::unzip (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E3A06: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E7126: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2E36F0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E64EB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x28F6D1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x296275: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 2,008 bytes in 1 blocks are still reachable in loss record 45 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x2F197E: crossbeam_epoch::sync::queue::Queue<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2F2202: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2F1ECD: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E9902: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E1206: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E4E49: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E2F5A: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E59F0: _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hd13678e4d803656bE.llvm.9549009445676040077 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E03BE: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x73F249: call_once<(),FnOnce<()>> (boxed.rs:1081)
==13771==    by 0x73F249: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==13771==    by 0x73F249: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==13771== 
==13771== 4,096 bytes in 4 blocks are still reachable in loss record 46 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x2E9740: crossbeam_deque::deque::Worker<T>::new_lifo (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2EA5D7: core::iter::traits::iterator::Iterator::unzip (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E3A06: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E7126: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x729FA9: std::sync::once::Once::call_inner (once.rs:416)
==13771==    by 0x2E36F0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E64EB: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x28F6D1: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x296275: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x29AA28: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2B6471: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771== 
==13771== 6,080 bytes in 4 blocks are still reachable in loss record 47 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x2E2928: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E59F0: _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hd13678e4d803656bE.llvm.9549009445676040077 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E03BE: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x73F249: call_once<(),FnOnce<()>> (boxed.rs:1081)
==13771==    by 0x73F249: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==13771==    by 0x73F249: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==13771==    by 0x52516DA: start_thread (pthread_create.c:463)
==13771==    by 0x57A271E: clone (clone.S:95)
==13771== 
==13771== 8,160 bytes in 4 blocks are still reachable in loss record 48 of 48
==13771==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13771==    by 0x2EF25F: crossbeam_epoch::internal::Local::register (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2F1E54: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E9902: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E1206: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E4E49: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E2F5A: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E59F0: _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hd13678e4d803656bE.llvm.9549009445676040077 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x2E03BE: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/memory)
==13771==    by 0x73F249: call_once<(),FnOnce<()>> (boxed.rs:1081)
==13771==    by 0x73F249: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==13771==    by 0x73F249: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==13771==    by 0x52516DA: start_thread (pthread_create.c:463)
==13771==    by 0x57A271E: clone (clone.S:95)
==13771== 
==13771== LEAK SUMMARY:
==13771==    definitely lost: 432 bytes in 1 blocks
==13771==    indirectly lost: 232 bytes in 5 blocks
==13771==      possibly lost: 1,240 bytes in 5 blocks
==13771==    still reachable: 30,530 bytes in 103 blocks
==13771==         suppressed: 0 bytes in 0 blocks
==13771== 
==13771== For counts of detected and suppressed errors, rerun with: -v
==13771== ERROR SUMMARY: 4 errors from 4 contexts (suppressed: 0 from 0)
